concent_api/core/subtask_helpers.py

```diff
@@ -185,3 +188,15 @@ def are_subtask_results_accepted_messages_signed_by_the_same_requestor(subtask_r
         ) for subtask_results_accepted in subtask_results_accepted_list
     )
     return are_all_signed_by_requestor
+
+
+def get_one_or_none(
+    query_set_or_model: Union[django.db.models.Model, django.db.models.QuerySet],
+    **conditions: Any
+)-> Optional[django.db.models.Model]:
+    if isinstance(query_set_or_model, django.db.models.base.ModelBase):
+        instances = query_set_or_model.objects.filter(**conditions)
+        return None if len(instances) == 0 else instances[0]
+    else:
+        instances = query_set_or_model.filter(**conditions)
+        return None if len(instances) == 0 else instances[0]
```

Comment:
This function is meant to be used only in cases when there's at most one instance.

You should add an `assert` to enforce that.
``` python
assert len(instances) <= 1
```

BTW, any code that gets the first element from a collection (e.g. `instances[0]`) should be a big, red flag to you. If you take element `[0]` always ask yourself - what if there's more than one element? Usually the answer is that there can be only one element and you should add an `assert` to make that clear.

Reply:
Right, I thought about this, what will happen if it will be more then one element.. But I agree with you, this function is to return one or none, so it should be used properly. Done